### ISSUE TITLE
Question : how to debug the failure ?

### ISSUE THREAD

**Issue by FrancoisH** [Dymantic/laravel-instagram-feed#12]
Sorry to ask again but, I did make it work correctly on my local computer.

I did deploy it on my distant server, and added the correct Callback URI into the Facebook / Instagram App. There's no error on the Instagram authentication process side, but I get a failure on the callback each time... even I did set it up exactly the same way as I did on the local development environment ...

With only my failure as a return, how can I debug? Thanks.

**Comment by michaeljoyner** [comment]
What error/failure are you getting? What response comes back from instagram?

**Comment by FrancoisH** [comment]
I don't have any error visible, just the redirect to my failure page.

**Comment by FrancoisH** [comment]
```{"error_type": "OAuthException", "code": 400, "error_message": "This authorization code has been used"}```

I don't understand how it can have been used already?

**Comment by michaeljoyner** [comment]
That is odd. There is a refreshToken method on the profile, see if you can use that. I should add some better debugging options.

**Comment by FrancoisH** [comment]
``` here is the real error : GET https://graph.instagram.com/1.7841400774391E+16?fields=id,username&access_token=IGQVJWSXc4elEyd2JUSDZA4V0FSNzhDQU1hUjJJdUFPNGlDUHJObEpZAMnNOU2paa0VFbzV2eTVENGxwcW9BNVdhaDRibDFuSFhEdllsWEFyOUNvYm8zTUFuZAXVuSlVfZA2E3YWo4UUhlNG9pbnh6TjZArQURPYkQxbUpVMmow` resulted in a `400 Bad Request` response:
{"error":{"message":"Unsupported get request. Object with ID '1.7841400774391E+16' does not exist, cannot be loaded due  (truncated...)
 {"exception":"[object] (Dymantic\\InstagramFeed\\Exceptions\\AccessTokenRequestException(code: 0): Client error: `GET https://graph.instagram.com/1.7841400774391E+16?fields=id,username&access_token=IGQVJWSXc4elEyd2JUSDZA4V0FSNzhDQU1hUjJJdUFPNGlDUHJObEpZAMnNOU2paa0VFbzV2eTVENGxwcW9BNVdhaDRibDFuSFhEdllsWEFyOUNvYm8zTUFuZAXVuSlVfZA2E3YWo4UUhlNG9pbnh6TjZArQURPYkQxbUpVMmow` resulted in a `400 Bad Request` response:
{\"error\":{\"message\":\"Unsupported get request. Object with ID '1.7841400774391E+16' does not exist, cannot be loaded due  (truncated...)
 at /home/likeabcogh/www-v2/vendor/dymantic/laravel-instagram-feed/src/Profile.php:49)```

**Comment by FrancoisH** [comment]
Why does it get the ID as a float ???

**Comment by michaeljoyner** [comment]
Not sure, but certainly shouldn't be. Maybe clear your profile and start over? I can't really did too much while I'm away from my computer.

**Comment by FrancoisH** [comment]
Already did that... There's no reason for the return from Instagram being different from the two setups except if there's a different curl version on the server... Or a PHP flag for hash parameters...  I don't see any other possibility as it work like a charm on local computer.

**Comment by michaeljoyner** [comment]
For some reason the user_id stored work the token is being treated as a float and not a string.

**Comment by FrancoisH** [comment]
Hello there,

I did find a workaround... I decided to dive in and make my own implementation which is working quite nicely...

I encountered the same issue with the ID being converted to float... It's linked to `json_decode`...

I had to use this : `json_decode($response->getBody(), false, 512, JSON_BIGINT_AS_STRING)` and miracle... this is working as intended now.

This is only happening if your server have an older version of PHP (it occur in certain releases of 7.3) or with certain legacy options activated which is the case on this shared web host.

Thanks.
